README.md

```diff
@@ -0,0 +1,136 @@
+```
+
+> [!IMPORTANT]
+> In order to minimize unexpected collisions and preemption, tools installed by backplane-tools are not added to your $PATH by default.
+
+### Install everything
+```shell
+backplane-tools install all
+```
+or
+```shell
+backplane-tools install
+```
+
+### Install a specific thing
+```shell
+backplane-tools install <tool name>
+```
+
+### Upgrade everything
+```shell
+backplane-tools upgrade all
+```
+
+> [!WARNING]
+> Using `backplane-tools upgrade all` is the same as running `backplane-tools install all`. See the [Upgrading](#upgrading) design section for more details.
+
+### Upgrade a specific thing
+```shell
+backplane-tools upgrade <tool name>
+```
+
+### Remove everything
+```shell
+backplane-tools remove all
+```
+
+### Remove a specific thing
+```shell
+backplane-tools remove <tool name>
+```
+
+## Design
+
+backplane-tools strives to be simplistic and non-invasive; it should not conflict with currently installed programs, nor should it require extensive research before operating.
+
+### Directory structure
+
+The following diagram summarizes the directory structure backplane-tools generates when managing tools on the local filesystem:
+```
+                                    $HOME/
+                                      |
+                                      V
+                                   .local/
+                                      |
+                                      V
+                                     bin/
+                                      |
+                                      V
+                                  backplane/
+                                      |
+            -------------------------------------------------------------
+           |                          |                                  |
+           V                          V                                  V
+        latest/                     toolA/                             toolB/ 
+           |                          |                                  |
+      ---------             ---------------------              ----------------//--
+     |         |           |                     |            |          |         |
+     V         V           V                     V            V          V         V
+   linkToA linkToB     version1/             version2/    version0.1/   ...    version10.0/
+                           |                     |            |                    |
+                           V                     V            V                    V
+                       ---------             ---------       ...              -----------
+                      |         |           |         |                      |           |
+                      V         V           V         V                      V           V
+                  executable  README    executable* README                executable*  README
+
+* = linked to the latest/ directory
+```
```

Comment:
I like this diagram, but fyi you can also use mermaid to generate Markdown diagrams, like https://github.com/openshift/custom-domains-operator/blob/b9b8893a86c27f5a24b10e32f6a20894cc7b5a0c/README.md#L8-L18

Reply:
Plus one for Mermaid. It's super handy.

Reply:
Yeah, I opted for an ascii diagram initially, just to keep things simple. I also like that it's readable regardless if you're viewing the compiled README or the raw text, but wouldn't be opposed if someone decided to update this chart in future iterations

Reply:
I agree, having it viewable in raw text mode is a nice bonus.